lib/pushmi_pullyu/aip/downloader.rb

```diff
@@ -1,215 +1,184 @@
-                                             optional: path_spec.optional,
-                                             rdf: rdf)
-    log_saved(path_spec, success, local_path: output_file)
+    is_success = fedora_fetcher.download_object(output_file,
+                                                url_extra: path_spec.remote,
+                                                optional: path_spec.optional,
+                                                is_rdf: is_rdf)
+    log_saved(is_success, output_file)
   end
 
   def download_permissions
     PushmiPullyu.logger.info("#{@noid}: looking up permissions from Solr ...")
     results = PushmiPullyu::AIP::SolrFetcher.new(@noid).fetch_permission_object_ids
     if results.empty?
       PushmiPullyu.logger.info("#{@noid}: permissions not found")
-      return
-    end
-    results.each do |permission_id|
-      PushmiPullyu.logger.info("#{@noid}: permission object #{permission_id} found")
-      download_permission(permission_id)
+    else
+      results.each do |permission_id|
+        PushmiPullyu.logger.info("#{@noid}: permission object #{permission_id} found")
+        download_permission(permission_id)
+      end
     end
   end
 
   def download_permission(permission_id)
-    permission_fetcher =
-      PushmiPullyu::AIP::FedoraFetcher.new(permission_id)
-
-    filename = permission_filename(permission_id)
-
     path_spec = OpenStruct.new(
       remote: nil,
-      local: filename,
+      local: "#{aip_dirs.metadata}/permission_#{permission_id}.n3",
       optional: false
     )
-    download_and_log(path_spec, fedora_fetcher: permission_fetcher)
+    download_and_log(path_spec, PushmiPullyu::AIP::FedoraFetcher.new(permission_id))
   end
 
   ### Logging
 
-  def log_fetching(path_spec, url, local_path: nil)
-    local_path ||= path_spec.local
-    message = "#{@noid}: #{local_path} -- fetching from #{url} ..."
-    PushmiPullyu::Logging.log_aip_activity(@noid, message)
+  def log_fetching(url, output_file)
+    message = "#{@noid}: #{output_file} -- fetching from #{url} ..."
+    PushmiPullyu::Logging.log_aip_activity(aip_directory, message)
   end
 
-  def log_saved(path_spec, success = true, local_path: nil)
-    local_path ||= path_spec.local
-    message = if success
-                "#{@noid}: #{local_path} -- saved"
+  def log_saved(is_success, output_file)
+    message = if is_success
+                "#{@noid}: #{output_file} -- saved"
               else
-                "#{@noid}: #{local_path} -- not_found"
+                "#{@noid}: #{output_file} -- not_found"
               end
-    PushmiPullyu::Logging.log_aip_activity(@noid, message)
+    PushmiPullyu::Logging.log_aip_activity(aip_directory, message)
   end
 
   ### Directories
 
   def aip_dirs
     @aip_dirs ||= OpenStruct.new(
-      objects: 'objects',
-      metadata: 'objects/metadata',
-      logs: 'logs',
-      thumbnails: 'thumbnails'
+      objects: "#{aip_directory}/data/objects",
```

Comment:
we explode these via a `full_local_path` method in many places throughout this code. Just do it here once and don't have to worry about it again.